tools/tpm2_getmanufec.c

```diff
@@ -582,13 +526,19 @@ static bool on_option(char key, char *value) {
         ctx.ek_server_addr = value;
         break;
     case 'i':
-        return_val = tpm2_util_string_to_uint32(value, &ctx.auth_session_handle);
+        return_val = tpm2_util_string_to_uint32(value, &ctx.endorse_session_data.sessionHandle);
+        if (!return_val) {
+            LOG_ERR("Could not convert session handle to number, got: \"%s\"",
+                    value);
+            return false;
+        }
+
+        return_val = tpm2_util_string_to_uint32(value, &ctx.owner_session_data.sessionHandle);
```

Comment:
This could be a lot more compact:
```
UINT32 tmp;
return_val = tpm2_util_string_to_uint32(value, &tmp);
if (error)...
fi
ctx.owner_session_data.sessionHandle =
ctx.endorse_session_data.sessionHandle = tmp;
```
or
```
return_val = tpm2_util_string_to_uint32(value, &ctx.endorse_session_data.sessionHandle)
                    && tpm2_util_string_to_uint32(value, &ctx.owner_session_data.sessionHandle);
if (error)....
fi
```

Reply:
indeed, I wasn't thinking clearly.